backend/routes/user.routes.ts

```diff
@@ -0,0 +1,59 @@
+import express from "express";
+const router = express.Router();
+const passport = require("passport");
+const bcrypt = require("bcrypt");
+const User = require("../models/User");
+
+import { DatabaseUserInterface } from "../utils/Interfaces";
+
+router.post("/signup", async (req, res) => {
+	const { username, password } = req?.body;
+	if (
+		!username ||
+		!password ||
+		typeof username !== "string" ||
+		typeof password !== "string"
+	) {
+		res.send("Improper Values");
+		return;
+	}
+	User.findOne(
+		{ username },
+		async (err: Error, userExist: DatabaseUserInterface) => {
+			if (err) throw err;
+			if (userExist) res.send("User Already Exists");
+			if (!userExist) {
+				const hashedPassword = await bcrypt.hash(password, 10);
+				const newUser = new User({
+					username,
+					password: hashedPassword,
+				});
+				await newUser.save();
+				res.send("success");
+			}
+		}
+	);
+});
+
+router.post("/login", passport.authenticate("local"), (req, res) => {
+	res.send("success");
+});
+
+router.get("/user", (req, res) => {
+	res.send(req.user);
+});
+
+router.post("/delete", async (req, res) => {
+	const { id } = req?.body;
```

Comment:
Why do you have here `?`.
I assume that the user won't be able to hit this route if there is no idea. That is something that is being checked on the front-end. Or do you expect that the `req` doesn't have a body?

(Because now you're checking if `req` has the body property, which is always the case. I actually also assume that you want to check if there is a `id`)